server/chat-plugins/helptickets.js

```diff
@@ -758,6 +848,165 @@ const pages = {
 			buf += `</tbody></table></div>`;
 			return buf;
 		},
+		stats(query, user, connection) {
+			// view-help-stats-TABLE-YYYY-MM-COL
+			if (!user.named) return Rooms.RETRY_AFTER_LOGIN;
+			this.title = 'Ticket Stats';
+			if (!this.can('lock')) return;
+
+			let table = query.shift() || '';
+			if (!['staff', 'tickets'].includes(table)) table = 'tickets';
+			let monthString = `${query.shift()}-${query.shift()}`;
+			let date = null;
+			if (monthString.includes('undefined')) {
+				// year/month not provided, use current month
+				date = new Date();
+				monthString = Chat.toTimestamp(date).split(' ')[0].split('-', 2).join('-');
+			}
+			let rawTicketStats = FS(`logs/tickets/${monthString}.csv`).readIfExistsSync();
+			if (!rawTicketStats) return `<div class="pad"><br />No ticket stats found.</div>`;
+
+			// Calculate next/previous month for stats and validate stats exist for the month
+			const prevDate = new Date(monthString);
+			if (!date) date = new Date((prevDate.getMonth() === 11 ? prevDate.getFullYear() + 1 : prevDate.getFullYear()), (prevDate.getMonth() === 11 ? 0 : prevDate.getMonth() + 1));
+			const nextDate = new Date((prevDate.getMonth() >= 10 ? prevDate.getFullYear() + 1 : prevDate.getFullYear()), (prevDate.getMonth() + 2 > 11 ? (prevDate.getMonth() + 2) - 12 : prevDate.getMonth() + 2));
+			let prevString = Chat.toTimestamp(prevDate).split(' ')[0].split('-', 2).join('-');
+			let nextString = Chat.toTimestamp(nextDate).split(' ')[0].split('-', 2).join('-');
+
+			let buttonBar = '';
+			if (FS(`logs/tickets/${prevString}.csv`).readIfExistsSync()) {
```

Comment:
Why are we reading in the whole file just to check if it exists :O? just `lstat`??

Reply:
What about windows/mac support? Also would it potentially be possible to implement something on FS to check if a file exists without reading it maybe? @Zarel